src/library/scala/collection/immutable/ArraySeq.scala

```diff
@@ -231,7 +231,7 @@ object ArraySeq extends StrictOptimizedClassTagSeqFactory[ArraySeq] { self =>
     override def hashCode = MurmurHash3.arraySeqHash(unsafeArray)
     override def equals(that: Any): Boolean = that match {
       case that: ofRef[_] =>
-        BoxesRunTime.arraysEquals(
+        Array.equals(
```

Comment:
Why not just reuse `java.util.Arrays.equals(Object[] a, Object[] a2)` ?

Reply:
That doesn't honour Scala's collaborative equality. Or cooperative equality. Whatever it's called.

Anyways, this PR is just about moving the function.

Reply:
BTW, it also safe in comparing of uncomparable stuff, while Java version still has a bug:
```
$ scala
Welcome to Scala 2.12.4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_171).
Type in expressions for evaluation. Or try :help.

scala> Float.NaN != Float.NaN
res0: Boolean = true

scala> !Float.NaN.equals(Float.NaN)
res1: Boolean = false
```

Reply:
Ups, looks like it doesn't work for corner case of uncomparable stuff:
```
$ scala
Welcome to Scala 2.12.4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_171).
Type in expressions for evaluation. Or try :help.

scala> java.lang.Float.valueOf(Float.NaN) == java.lang.Float.valueOf(Float.NaN)
res0: Boolean = false

scala> java.lang.Float.valueOf(Float.NaN) != java.lang.Float.valueOf(Float.NaN)
res1: Boolean = true

scala> :paste
// Entering paste mode (ctrl-D to finish)

object ArrayTest {
def equals(xs: Array[AnyRef], ys: Array[AnyRef]): Boolean = {
    if (xs eq ys)
      return true
    if (xs.length != ys.length)
      return false

    val len = xs.length
    var i = 0
    while (i < len) {
      if (xs(i) != ys(i))
        return false
      i += 1
    }
    true
  }
}

// Exiting paste mode, now interpreting.

defined object ArrayTest

scala> val a = Array[AnyRef](java.lang.Float.valueOf(Float.NaN))
a: Array[AnyRef] = Array(NaN)

scala> ArrayTest.equals(a, a)
res2: Boolean = true

scala> a(0) == a(0)
res3: Boolean = true
```

Reply:
@plokhotnyuk This is a followup to #7027. Please see that, and the issue it fixes, for details.